jekyll/_posts/guides/2018-09-21-usage-of-the-matrix-js-sdk.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: post
+categories: guides
+---
+
+<style>
+ul {
+    list-style: circle;
+    margin-left: 30px;
+}
+p {
+    padding-bottom: 1em;
+    line-height: 1.7em;
+}
+h2 {
+    margin-bottom: 15px;
+    margin-top: 30px;
+}
+
+</style>
+
+# Usage of the matrix-js-sdk
+
+Let's explore how we would make a very simple Matrix client, with only the ability to perform an initial sync, and to get member lists and the timeline for rooms of our choice.
+
+This article will explore the [Matrix Client-Server API], making use of the [matrix-js-sdk]. Later articles may discuss making the underlying calls. Specifically we will cover:
+
+* login
+* simple syncing
+* listening for timeline events
+* access the `MatrixInMemoryStore`
+* sending messages to rooms
+
+We'll use Node.js as our environment, though the [matrix-js-sdk] can also be used directly in the browser.
+
+## Let's write JavaScript
+
+The first thing to do is install the [matrix-js-sdk] from NPM:
+
+```unix
+npm install matrix-js-sdk
+```
+
+We include the SDK in our source exactly as expected:
+
+```javascript
+var sdk = require("matrix-js-sdk");
+```
+
+## Login with an access token
+
+Instantiate a new client object and use an `access token` to login:
+
+```javascript
+var client = sdk.createClient({
+    baseUrl: "https://matrix.org",
+    accessToken: "....MDAxM2lkZW50aWZpZXIga2V5CjAwMTBjaWQgZ2Vu....",
+    userId: "@USERID:matrix.org"
+});
+// note that we use the full MXID for the userId value
+```
+
+If you are logged into Riot, you can find an `access token` for the logged-in user on the Settings page.
+
+We can also retrieve an `access token` programmatically using the API:
+
+```javascript
+var client = sdk.createClient("https://matrix.org");
+client.login("m.login.password", {"user": "USERID", "password":"hunter2"}, function(err, response){
+    console.log(response.access_token);
+});
+```
+
+In any case, once logged in either with a password or an access token, the client can get the current access token via:
+
+```javascript
+console.log(client.getAccessToken());
+```
+
+## Sync and Listen
+
+Let's perform a first sync, and listen for the response:
+
+```javascript
+client.once('sync', function(state, prevState, res) {
+    //console.log(res);
+    if(state === 'PREPARED') {
+        console.log("prepared");
+    } else {
+        console.log(state);
+        process.exit(1);
```

Comment:
It's not unheard of for us to add intermediate states to the js-sdk, in which case this code would break - it's probably better to ignore any other state changes.